src/python/pants/backend/native/subsystems/libc_dev.py

```diff
@@ -99,11 +99,8 @@ def _host_libc(self):
 
     return self._get_host_libc_from_host_compiler()
 
-  def get_libc_dirs(self, platform):
+  @memoized_method
+  def get_libc_objects(self):
     if not self.get_options().enable_libc_search:
       return []
-
-    return platform.resolve_platform_specific({
-      'darwin': lambda: [],
-      'linux': lambda: [self._host_libc.get_lib_dir()],
-    })
+    return [self._host_libc.crti_object]
```

Comment:
Consider the ternary form since it fits on one line and - more importantly - places the positive case 1st:
```python
return [self._host_libc.crti_object] if self.get_options().enable_libc_search else []
```

Reply:
Yes, this snippet was particularly crooked. I have made the change, and have heard the note on the positive case 1st.